src/esockd_proxy_protocol.erl

```diff
@@ -98,6 +90,31 @@ recv(Transport, Sock, Timeout) ->
             {error, proxy_proto_timeout}
     end.
 
+recv_v2(Transport, Sock, OriginOpts, Deadline) ->
+    Transport:setopts(Sock, [{active, false}, {packet, raw}]),
+    with_remaining_timeout(Deadline, fun(HeaderTimeout) ->
+        case Transport:recv(Sock, 14, HeaderTimeout) of
+            {ok, <<?SIG, 2:4, Cmd:4, AF:4, Trans:4, Len:16>>} ->
+                with_remaining_timeout(Deadline, fun(ProxyInfoTimeout) ->
+                    case Transport:recv(Sock, Len, ProxyInfoTimeout) of
+                        {ok, ProxyInfo} ->
+                            Transport:setopts(Sock, OriginOpts),
```

Comment:
Maybe we should add this to an `after` clause that always restores the original opts in the event of failures?

Reply:
I thought that the idea was that the only safe and possible action on an erroneous (closed/etc) socket is to close it, so I just followed the pattern.

Reply:
this makes me winder if we want to setopts on failure at all.
if parse_v2 returns error or crashes, the socket is to be closed immediately anyway.
so maybe setopts after receive_v2 returns ok.
assumption: the delay of parse_v2 should not impact performance.